packages/salesforce-adapter/src/adapter.ts

```diff
@@ -390,17 +388,18 @@ export default class SalesforceAdapter implements AdapterOperations {
     )
     const metadataInstances = this.fetchMetadataInstances(metadataTypeInfos, metadataTypes)
 
-    const elements = _.flatten(
-      await Promise.all([annotationTypes, fieldTypes, missingTypes,
-        metadataTypes]) as Element[][]
-    )
+    const elements = [
+      ...annotationTypes, ...fieldTypes, ...missingTypes, ...(await metadataTypes),
+    ] as Element[]
+    progressReporter.reportProgress({ details: 'Fetched all types. Reading instances' })
 
     const {
       elements: metadataInstancesElements,
       configChanges: metadataInstancesConfigInstances,
     } = await metadataInstances
     elements.push(...metadataInstancesElements)
 
+    progressReporter.reportProgress({ details: 'Running filters on the elements' })
```

Comment:
Just to be consistent with the previous message
```suggestion
    progressReporter.reportProgress({ details: 'Finished fetching instances. Running filters for additional information' })
```